### DISCUSSION TITLE
Changing Ports causes the program to not search tidal

### DISCUSSION THREAD

**Discussion by yamimizu832** [RandomNinjaAtk/docker-lidarr-extended#224] [Q&A] [answered]
Hello. Below is my compose file for my instance.

version: "2.1"
services:
  lidarr-extended:
    image: randomninjaatk/lidarr-extended:latest
    container_name: lidarr-extended-320
    volumes:
      - /home/ravanafr/Lidarr-320/config:/config
      - /home/ravanafr/Lidarr-320/downloads:/downloads-lidarr-extended
      - /home/ravanafr/instantup/Lidarr320:/music
      - /home/ravanafr/Lidarr-320/music videos:/music-videos
    environment:
      - TZ=America/New_York
      - PUID=1000
      - PGID=1000
      - enableAudioScript=true
      - enableVideoScript=true
      - scriptInterval=15m
      - configureLidarrWithOptimalSettings=true
      - searchSort=date
      - audioFormat=native
      - audioBitrate=high
      - requireQuality=true
      - enableReplaygainTags=true
      - audioLyricType=both
      - dlClientSource=tidal
      - arlToken=Token_Goes_Here
      - tidalCountryCode=US
      - addDeezerTopArtists=true
      - addDeezerTopAlbumArtists=false
      - addDeezerTopTrackArtists=false
      - topLimit=10
      - addRelatedArtists=false
      - numberOfRelatedArtistsToAddPerArtist=5
      - lidarrSearchForMissing=true
      - addFeaturedVideoArtists=false
      - plexUrl=http://x.x.x.x:32400
      - plexToken=Token_Goes_Here
      - youtubeSubtitleLanguage=en
      - enableQueueCleaner=true
      - matchDistance=5
      - enableBeetsTagging=true
      - beetsMatchPercentage=90
      - retryNotFound=90
    ports:
      - 5686:5686
    restart: unless-stopped

I would like to run this program on port 5686 instead of 8686 since I have a separate instance of lidarr already running on that port. For some reason changing the port in the compose causes the audio_sh and video_sh scripts not to start. I spent an hour trying to figure the issue and I am stumped. If I disable my first instance of lidarr that is not using this extension and then change the port within the compose file to 8686 everything works perfectly. However when it is changed to 5686 it wont trigger the scripts to run. Port 5686 is open and working. Is there some additional step that I need to take? Also in the compose file I define the port, however, when the config.xml is created on the containers first startup it automatically assigned the port to 8686 instead of the defined 5686 in the compose file. I can manually edit it to change it to 5686 but is there a reason it does this?

**Comment by RandomNinjaAtk** [comment]
Sounds like a misconfiguration, the internal port should never need to be change, you should only be changing the exposed port....

So what I believe you want is

`- 5686:8686`


Edit:
Also, if you changed the port, and did not update the lidarr web interface (settings) with the new port, that is also likely your problem...

**Reply by yamimizu832** [reply]
So the issue is when i list the port as 5686 in the compose the config.xml file that is generated when the container is started will show the the port as 8686. Since there is already another lidarr instance running on 8686 I cannot access this containers lidarr. So how would I go about being able to access this lidarr outside of port 8686?

**Reply by RandomNinjaAtk** [reply]
Port mappings allow the host to see one port and the container to use another internally.

Which is what I suggested. The container does have to be running in bridge mode, not host. Otherwise mappings are ignored. You can look online for more information. But I run 2 setups using port mapping features, on an unraid host system. So I'm. It using docker compose to give you more specifics.

**Reply by RandomNinjaAtk** [reply, marked as answer]
Didn't realize it, but I already have it explained in the readme, thanks to linuxserver.io's documentation. Read the Parameters description before the table.... https://github.com/RandomNinjaAtk/docker-lidarr-extended#parameters

Edit:
So what I told you previously `- 5686:8686` should work if you modify your compose file...

**Reply by yamimizu832** [reply]
Understood. Thank you for the explanation. I've never had to use port mapping. Since both instances will be mapped to the same port does that mean you will only need to re authenticate to tidal on one instance and both will have access to download from them?

Also I did not realize that 2 things could be mapped to one port so the parameter in the readme didnt click in my mind.

**Reply by RandomNinjaAtk** [reply]
Each container is it's own separate instance, so you would have to authenticate both of them, if you were running more than one instance.

**Reply by yamimizu832** [reply]
Got it. Thank you for explanations and the development of this tool. I appreciate your support
